pinot-core/src/main/java/com/linkedin/pinot/core/segment/creator/ColumnIndexCreationInfo.java

```diff
@@ -24,44 +24,19 @@
 
 public class ColumnIndexCreationInfo {
   private final boolean createDictionary;
-  private final Object min;
-  private final Object max;
-  private final Object sortedUniqueElementsArray;
   private final ForwardIndexType forwardIndexType;
   private final InvertedIndexType invertedIndexType;
-  private boolean isSorted;
-  private final boolean hasNulls;
-  private final int totalNumberOfEntries;
-  private final int maxNumberOfMultiValueElements;
-  private final int lengthOfShortestEntry;
-  private final int lengthOfLongestEntry;
   private final boolean isAutoGenerated;
-  private final PartitionFunction partitionFunction;
-  private final List<IntRange> partitionRanges;
   private final Object defaultNullValue;
-  private int numPartitions;
+  private final ColumnStatistics columnStatistics;
 
-  public ColumnIndexCreationInfo(boolean createDictionary, Object min, Object max, Object sortedUniqueElementsArray,
```

Comment:
That probably has to be the constructor with most args that I've seen. Glad we reduced that!